src/ape/contracts/base.py

```diff
@@ -153,6 +153,8 @@ def __repr__(self) -> str:
         return self.abi.signature
 
     def serialize_transaction(self, *args, **kwargs) -> TransactionAPI:
+        if "sender" in kwargs and isinstance(kwargs["sender"], ContractInstance):
+                kwargs["sender"] = self.account_manager.test_accounts[kwargs["sender"].address]
```

Comment:
I wonder if defaults are not getting set because this is no longer an `AccountAPI`.
We may need to move this logic some place else.

Reply:
yes I think this is the case, since it is only happening on the impersonation from a contract from what I can tell.

Reply:
https://github.com/ApeWorX/ape-hardhat/pull/44 is passing tests however when trying to impersonate a contract I am still getting `ERROR: (SignatureError) The transaction is not signed.`

Reply:
https://github.com/ApeWorX/ape-hardhat/pull/48